app/components/c-panel.vue

```diff
@@ -0,0 +1,419 @@
+            'gray'
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines if the triangle shaped button shall be displayed or not
+       */
+      triangleButton: {
+        type: Boolean,
+        default: false
+      },
+
+      /**
+       * Defines if the triangle shaped button is initially open or not
+       */
+      triangleButtonOpen: {
+        type: Boolean,
+        default: false
+      },
+
+      /**
+       * Defines the size of the triangle button
+       *
+       * Valid values: `small`, `big`
+       */
+      triangleButtonSize: {
+        type: String,
+        default: 'small',
+        validator(value) {
+          return [
+            'small',
+            'big'
+          ].includes(value);
+        },
+      },
+
+      /**
+       * Defines the plus color of the plus for the triangle button
+       *
+       * Valid values: `white`, `blue`
+       */
+      triangleButtonPlusColor: {
+        type: String,
+        default: 'white',
+        validator(value) {
+          return [
+            'white',
+            'blue'
+          ].includes(value);
+        },
+      },
+
+    },
+    // data() {
+    //   return {};
+    // },
+
+    computed: {
+      hasHeading() {
+        return this.heading;
+      },
+      panelModifiers() {
+        return {
+          border: this.border.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+          color: this.color,
+        };
+      },
+      headingModifiers() {
+        return {
+          spacing: this.headingSpacing.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+        };
+      },
+      contentModifiers() {
+        return {
+          spacing: this.contentSpacing.toString(), /* TODO - remove .toString() once vue-bem-cn accepts numbers */
+        };
+      },
+      triangleButtonModifiers() {
+        return {
+          size: this.triangleButtonSize,
+        };
+      },
+      shadowModifiers() {
+        return {
+          color: this.color,
+        };
+      },
+      plusModifiers() {
+        return {
+          size: this.triangleButtonSize,
+          color: this.triangleButtonPlusColor,
+          open: this.triangleButtonOpen,
+        };
+      },
+    },
+    // watch: {},
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() { },
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      onPlusToggle() {
+        // onPlusToggle() gets triggered on touch devices as well, which is not what we want.
+        if (!this.hasTouch()) {
+          this.$refs.plus.classList.toggle('c-panel__plus--hover');
+          this.$refs.shadow.classList.toggle('c-panel__shadow--hover');
+        }
+      },
+      onTriangleButtonClick() {
+        this.$emit('triangle-pressed');
+      },
+      hasTouch() {
```

Comment:
- [ ] I would recommend to move this to data(). I don't think the condition will change during runtime but the method will be called each time anyway.